src/app/pages/back-to-office/__tests__/back-to-office.page.spec.ts

```diff
@@ -1,31 +1,170 @@
+  let router: Router;
 
-  beforeEach(waitForAsync(() => {
+  configureTestSuite(() => {
     TestBed.configureTestingModule({
-      declarations: [BackToOfficePage],
-      imports: [IonicModule.forRoot()],
+      declarations: [
+        BackToOfficePage,
+        MockComponent(PracticeModeBanner),
+      ],
+      imports: [
+        IonicModule,
+        AppModule,
+        StoreModule.forRoot({
+          tests: () => ({
+            currentTest: {
+              slotId: '123',
+            },
+            testStatus: {},
+            startedTests: {
+              123: {
+                vehicleDetails: {},
+                accompaniment: {},
+                testData: {
+                  dangerousFaults: {},
+                  drivingFaults: {},
+                  manoeuvres: {},
+                  seriousFaults: {},
+                  testRequirements: {},
+                  ETA: {},
+                  eco: {},
+                  vehicleChecks: {
+                    showMeQuestion: {
+                      code: 'S3',
+                      description: '',
+                      outcome: '',
+                    },
+                    tellMeQuestion: {
+                      code: '',
+                      description: '',
+                      outcome: '',
+                    },
+                  },
+                  eyesightTest: {},
+                },
+                activityCode: '28',
+                journalData: {
+                  candidate: {
+                    candidateName: 'Joe Bloggs',
+                    driverNumber: '123',
+                  },
+                },
+                rekey: false,
+              },
+            },
+          }),
+        }),
+      ],
       providers: [
-        { provide: NavController, useClass: NavMock },
+        { provide: NavParams, useFactory: () => NavParamsMock.instance() },
+        { provide: Config, useFactory: () => ConfigMock.instance() },
+        { provide: Platform, useFactory: () => PlatformMock.instance() },
+        { provide: AuthenticationProvider, useClass: AuthenticationProviderMock },
+        { provide: DateTimeProvider, useClass: DateTimeProviderMock },
+        { provide: ScreenOrientation, useClass: ScreenOrientationMock },
+        { provide: Insomnia, useClass: InsomniaMock },
+        { provide: DeviceProvider, useClass: DeviceProviderMock },
         { provide: RouteByCategoryProvider, useClass: RouteByCategoryProviderMock },
       ],
-    }).compileComponents();
+    });
+  });
 
+  beforeEach(waitForAsync(() => {
     fixture = TestBed.createComponent(BackToOfficePage);
     component = fixture.componentInstance;
-    fixture.detectChanges();
+    screenOrientation = TestBed.inject(ScreenOrientation);
+    insomnia = TestBed.inject(Insomnia);
+    deviceProvider = TestBed.inject(DeviceProvider);
+    store$ = TestBed.inject(Store);
+    router = TestBed.inject(Router);
+    spyOn(store$, 'dispatch');
+    spyOn(router, 'navigate');
```

Comment:
can we remove this line, line 119 is also spying on navigate and returning a value